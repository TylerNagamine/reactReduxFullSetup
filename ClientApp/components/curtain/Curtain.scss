.curtain {
    height: 100%;
    width: 100%;
    background: black;
    position: absolute;
    opacity: .5;
}

.curtain-children {
    position: relative;
    top: 45%;
    text-align: center;
}