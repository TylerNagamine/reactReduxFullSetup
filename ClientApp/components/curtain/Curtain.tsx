import * as React from 'react';
import './Curtain.scss';

export interface ICurtainProps {
    children?: React.ReactNode;
    curtainText?: string;
    shown?: boolean;
}

export interface ICurtainState {
}

/**
 * Basic curtain component.  When shown is set, the curtain appears over the parent object.
 * Current limitation is that the parent must be set the position: relative.
 * Props: curtainText: string, shown: boolean
 */
export default class Curtain extends React.Component<ICurtainProps, ICurtainState> {
    constructor(props: ICurtainProps) {
        super(props);
    }

    public shouldComponentUpdate(nextProps: ICurtainProps, nextState: ICurtainState): boolean {
        return this.props.shown != nextProps.shown ||
            this.props.children != nextProps.children ||
            this.props.curtainText != nextProps.curtainText;
    }

    public render() {
        return (
            <div className={'curtain'} hidden={!this.props.shown}>
                <div className={'curtain-children'}>
                    {this.props.children}
                    <br />
                    <span style={{color: 'white'}}>{this.props.curtainText}</span>
                </div>
            </div>
        );
    }
}