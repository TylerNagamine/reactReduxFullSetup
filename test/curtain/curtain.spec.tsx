import {} from 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import Curtain from '../../ClientApp/components/curtain/Curtain';
import * as Renderer from 'react-test-renderer';

describe('Curtain component', () => {
    it('should match snapshot', () => {
        let shown = true;

        const component = Renderer.create(
            <Curtain 
                curtainText={"Test text"}
                shown={shown}
            />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render children', () => {
        const curtain = shallow(
            <Curtain shown={false}>
                <div className='unique'/>
            </Curtain>
        );

        expect(curtain.contains(<div className='unique'/>)).toBe(true);
    });
});