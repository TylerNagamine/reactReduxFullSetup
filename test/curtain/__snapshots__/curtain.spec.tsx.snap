// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Curtain component should match snapshot 1`] = `
<div
  className="curtain"
  hidden={false}
>
  <div
    className="curtain-children"
  >
    <br />
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      Test text
    </span>
  </div>
</div>
`;
